pkg/serving/config_changes.go

```diff
@@ -507,13 +507,13 @@ func GenerateVolumeName(path string) string {
 	builder := &strings.Builder{}
 	for idx, r := range path {
```

Comment:
Where is the code to guard against volume names matching the DNS label name restrictions 1, 3, and 4. That is the bullet points, from spec:

```text
Some resource types require their names to follow the DNS label standard as defined in RFC 1123. This means the name must:

1. contain at most 63 characters
2. contain only lowercase alphanumeric characters or '-'
3. start with an alphanumeric character
4. end with an alphanumeric character
```

There is clearly code to take care of bullet 2 but not the others.

Reply:
updated util to generate volume name to adhere to the requirements